openhands/server/routes/manage_conversations.py

```diff
@@ -269,19 +268,32 @@ async def search_conversations(
     running_conversations = await conversation_manager.get_running_agent_loops(
         user_id, conversation_ids
     )
-    connection_ids_to_conversation_ids = await conversation_manager.get_connections(filter_to_sids=conversation_ids)
-    result = ConversationInfoResultSet(
-        results=await wait_all(
```

Comment:
Hmm I believe `wait_all` was to parallelize information fetching instead of doing it one by one

Would be nice to double check the logic here, I think this may introduce a regression in performance 

cc @tofarr